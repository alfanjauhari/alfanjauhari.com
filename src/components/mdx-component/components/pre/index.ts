export * from './pre';
export * from './pre.styles';
