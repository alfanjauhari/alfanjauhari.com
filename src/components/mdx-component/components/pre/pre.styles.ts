import { styled } from '@/theme';

export const StyledPre = styled('pre', {
  my: '$5',
  backgroundColor: '$gray1',
  overflowX: 'auto',
  fontSize: '14px',
  lineHeight: 1.7,
  p: '$4',
  borderRadius: '5px',
});
