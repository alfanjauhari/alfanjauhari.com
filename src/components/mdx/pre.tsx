'use client'

import { CheckIcon, ClipboardIcon } from 'lucide-react'
import {
  type DetailedHTMLProps,
  type HTMLAttributes,
  useRef,
  useState,
} from 'react'

export default function Pre({
  children,
  ...props
}: DetailedHTMLProps<HTMLAttributes<HTMLPreElement>, HTMLPreElement>) {
  const [isCopied, setIsCopied] = useState(false)
  const preRef = useRef<HTMLPreElement>(null)

  const handleClickCopy = async () => {
    const code = preRef.current?.textContent

    if (code) {
      await navigator.clipboard.writeText(code)
      setIsCopied(true)

      setTimeout(() => {
        setIsCopied(false)
      }, 2000)
    }
  }

  return (
    <pre ref={preRef} {...props} className="relative">
      <button
        disabled={isCopied}
        onClick={handleClickCopy}
        className="absolute right-4 size-8 flex justify-center items-center rounded cursor-pointer hover:bg-gray-100 disabled:cursor-not-allowed"
        type="button"
      >
        {isCopied ? (
          <CheckIcon className="size-5 text-green-500" />
        ) : (
          <ClipboardIcon className="size-5 text-gray-500" />
        )}
      </button>
      {children}
    </pre>
  )
}
