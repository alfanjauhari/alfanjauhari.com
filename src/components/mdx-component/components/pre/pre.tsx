import { HTMLAttributes } from 'react';
import { StyledPre } from './pre.styles';

export type PreProps = HTMLAttributes<HTMLPreElement>;

export function Pre(props: PreProps) {
  return <StyledPre {...props} />;
}
