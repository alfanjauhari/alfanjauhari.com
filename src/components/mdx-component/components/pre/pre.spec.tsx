import { render, screen } from '@testing-library/react';
import { Pre } from './pre';

describe('MDXComponent/Pre', () => {
  it('should render successfuly', () => {
    render(<Pre data-testid="pre-component" />);

    expect(screen.getByTestId('pre-component')).toBeVisible();
  });

  it('should match snapshot', () => {
    render(<Pre data-testid="pre-component" />);

    expect(screen.getByTestId('pre-component')).toMatchSnapshot();
  });
});
