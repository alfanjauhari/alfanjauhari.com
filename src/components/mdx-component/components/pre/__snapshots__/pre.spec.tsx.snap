// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXComponent/Pre should match snapshot 1`] = `
<pre
  class="c-ieGfll"
  data-testid="pre-component"
/>
`;
